### PR TITLE
Reduce blackjax sampling memory usage

### PR DESCRIPTION
**Pull request by junpenglao**
## Description
Reduce blackjax sampling memory usage by not outputting the warm up diagnostics

## Related Issue

- [ ] Closes #
- [ ] Related to #

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [x] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc--7407.org.readthedocs.build/en/7407/

### CODE REVIEW

**Comment by junpenglao** [comment]
Will need to upgrade jaxlib requirement first (https://github.com/conda-forge/jaxlib-feedstock/pull/272)

**Comment by junpenglao** [comment]
Already addressed in #7311

File: pyproject.toml [hunk 1]
```diff
@@ -94,3 +94,11 @@ exclude_lines = [
...
 [tool.coverage.run]
 omit = ["*examples*"]
+
+filterwarnings =[
+  # JAX issues an over-eager warning if os.fork() is called when the JAX module is loaded, even if JAX isn't being used
+  'ignore:os\.fork\(\) was called\.:RuntimeWarning',
+
+  # Warning coming from blackjax
+  'ignore:jax\.tree_map is deprecated:DeprecationWarning',
+]
```
Comment (ricardoV94) [inline pyproject.toml]: This should be under [tool.pytest.ini_options] or in a pytest ini file if we use that here instead

Comment (junpenglao) [reply pyproject.toml]: Done, thanks

File: conda-envs/environment-jax.yml [hunk 2]
```diff
@@ -11,9 +11,9 @@ dependencies:
 - cloudpickle
 - h5py>=2.7
 # Jaxlib version must not be greater than jax version!
-- blackjax==1.2.0  # Blackjax>=1.2.1 is incompatible with latest available version of jaxlib in conda-forge
-- jaxlib==0.4.23  # Latest available version in conda-forge, update when new version is available
-- jax==0.4.23
+- blackjax
+- jaxlib
+- jax
```
Comment (ricardoV94) [inline conda-envs/environment-jax.yml]: We still need to pin to the same number no?

Comment (junpenglao) [reply conda-envs/environment-jax.yml]: My experience in blackjax is that the version is automatically resolved.

File: tests/sampling/test_mcmc_external.py [hunk 3]
```diff
@@ -51,7 +51,7 @@ def test_external_nuts_sampler(recwarn, nuts_sampler):
     warns = {
         (warn.category, warn.message.args[0])
         for warn in recwarn
-        if warn.category not in (FutureWarning, DeprecationWarning)
+        if warn.category not in (FutureWarning, DeprecationWarning, RuntimeWarning)
```
Comment (junpenglao) [inline tests/sampling/test_mcmc_external.py:54]: @ricardoV94 :see_no_evil:

Comment (ricardoV94) [reply tests/sampling/test_mcmc_external.py:54]: Beautiful

File: conda-envs/environment-jax.yml (hunk 2 above)
Comment (ricardoV94) [reply conda-envs/environment-jax.yml]: Can we lower pin the versions to the latest?

File: conda-envs/environment-jax.yml [hunk 4]
```diff
@@ -25,9 +25,7 @@ dependencies:
 - networkx
 - rich>=13.7.1
 - threadpoolctl>=3.1.0
-# JAX is only compatible with Scipy 1.13.0 from >=0.4.26, but the respective version of
-# JAXlib is still not on conda: https://github.com/conda-forge/jaxlib-feedstock/pull/243
-- scipy>=1.4.1,<1.13.0
+- scipy
```
Comment (ricardoV94) [inline conda-envs/environment-jax.yml]: Keep lower pin just because why not?

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**